### PR TITLE
Feat/request metadata

### PR DESCRIPTION
**Pull request by junminahn**
- add request metadata edit modal for admin.
-  fix minor UI warnings.

### CODE REVIEW

File: app/services/request.ts [hunk 1]
```diff
@@ -83,3 +83,12 @@ export const deleteRequest = async (id?: number): Promise<[Request[], null] | [n
     return handleAxiosError(err);
   }
 };
+
+export const updateRequestMetadata = async (data: Request): Promise<[Request, null] | [null, AxiosError]> => {
+  try {
+    const result = await instance.put('request-metadata', data).then((res) => res.data);
```
Comment (thegentlemanphysicist) [inline app/services/request.ts:89]: Naive question, is there a reason we don't have a get for the metadata? Or is the metadata a purely used for backend/metabase/database purposes?

**Review by thegentlemanphysicist — APPROVED** [review]
Looks good, I left a few questions about the code structure for my own education.  Any answers would be appreciated

File: app/services/request.ts (hunk 1 above)
Comment (junminahn) [reply app/services/request.ts:89]: oh, the API endpoint is to update integration metadata that requires a request body.
and `POST` is mostly for creating entity and `PUT` is to update, so used `PUT` in this case.

File: app/pages/_app.tsx [hunk 2]
```diff
@@ -24,9 +25,17 @@ const authenticatedUris = [
...
+
+  session.roles = session.client_roles;
+  session.isAdmin = session?.client_roles?.includes('sso-admin');
+  return session;
+};
+
 function MyApp({ Component, pageProps }: AppProps) {
   const router = useRouter();
-  const [currentUser, setCurrentUser] = useState(null);
+  const [currentUser, setCurrentUser] = useState<LoggedInUser | null>(null);
```
[the remark this answers is not in the document]
Comment (junminahn) [reply app/pages/_app.tsx:38]: yes, correct, once the frontend app gets the `id_token` then it validates the token and parses to get the necessary user information from the token.

File: app/page-partials/admin-dashboard/MetadataEditModal.tsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+import { Request } from 'interfaces/Request';
+import { workflowStatusOptions } from 'metadata/options';
+import { updateRequestMetadata } from 'services/request';
+
+interface Props {
+  request: Request;
+  onUpdate?: Function;
+}
+
+const StyledDropdown = styled(Dropdown)`
```
[the remark this answers is not in the document]
Comment (junminahn) [reply app/page-partials/admin-dashboard/MetadataEditModal.tsx:17]: good question, we tried to scope the types down to component level. and the library we use is `styled-components` which works okay. we can talk about the styling convention later together.

**Pull request merged**